org.lflang/src/org/lflang/federated/generator/FedPreambleEmitter.java

```diff
@@ -13,8 +14,16 @@ public FedPreambleEmitter() {}
      * Add necessary code to the source and necessary build support to
      * enable the requested serializations in 'enabledSerializations'
      */
-    String generatePreamble(FederateInstance federate, LinkedHashMap<String, Object> federationRTIProperties, ErrorReporter errorReporter) {
-        return FedTargetExtensionFactory.getExtension(federate.target).generatePreamble(federate, federationRTIProperties, errorReporter);
+    String generatePreamble(FederateInstance federate, FedFileConfig fileConfig, LinkedHashMap<String, Object> federationRTIProperties, ErrorReporter errorReporter)
+        throws IOException {
+        String pre = FedTargetExtensionFactory.getExtension(federate.target).generatePreamble(
+            federate, fileConfig, federationRTIProperties, errorReporter);
+        if (pre.length() == 0) return "";
+        else return
+        """
+        preamble {=
```

Comment:
Thanks! This looks great.

However, the common `preamble` syntax unfortunately doesn't work with the Cpp target because it requires a `public` or `private` modifier for the preamble. See [here](https://www.lf-lang.org/docs/handbook/preambles?target=cpp) for some examples.

Reply:
OK, then I think I'll revert this change and handle C & Python specifically then.

Reply:
Okay, thank you so much.